msgp/elsize_tinygo.go

```diff
@@ -0,0 +1,129 @@
+	case v == mbin8:
+		return bytespec{size: 2, extra: extra8, typ: BinType}
+	case v == mbin16:
+		return bytespec{size: 3, extra: extra16, typ: BinType}
+	case v == mbin32:
+		return bytespec{size: 5, extra: extra32, typ: BinType}
+	case v == mext8:
+		return bytespec{size: 3, extra: extra8, typ: ExtensionType}
+	case v == mext16:
+		return bytespec{size: 4, extra: extra16, typ: ExtensionType}
+	case v == mext32:
+		return bytespec{size: 6, extra: extra32, typ: ExtensionType}
+	case v == mfloat32:
+		return bytespec{size: 5, extra: constsize, typ: Float32Type}
+	case v == mfloat64:
+		return bytespec{size: 9, extra: constsize, typ: Float64Type}
+	case v == muint8:
+		return bytespec{size: 2, extra: constsize, typ: UintType}
+	case v == muint16:
+		return bytespec{size: 3, extra: constsize, typ: UintType}
+	case v == muint32:
+		return bytespec{size: 5, extra: constsize, typ: UintType}
+	case v == muint64:
+		return bytespec{size: 9, extra: constsize, typ: UintType}
+	case v == mint8:
+		return bytespec{size: 2, extra: constsize, typ: IntType}
+	case v == mint16:
+		return bytespec{size: 3, extra: constsize, typ: IntType}
+	case v == mint32:
+		return bytespec{size: 5, extra: constsize, typ: IntType}
+	case v == mint64:
+		return bytespec{size: 9, extra: constsize, typ: IntType}
+	case v == mfixext1:
+		return bytespec{size: 3, extra: constsize, typ: ExtensionType}
+	case v == mfixext2:
+		return bytespec{size: 4, extra: constsize, typ: ExtensionType}
+	case v == mfixext4:
+		return bytespec{size: 6, extra: constsize, typ: ExtensionType}
+	case v == mfixext8:
+		return bytespec{size: 10, extra: constsize, typ: ExtensionType}
+	case v == mfixext16:
+		return bytespec{size: 18, extra: constsize, typ: ExtensionType}
+	case v == mstr8:
+		return bytespec{size: 2, extra: extra8, typ: StrType}
+	case v == mstr16:
+		return bytespec{size: 3, extra: extra16, typ: StrType}
+	case v == mstr32:
+		return bytespec{size: 5, extra: extra32, typ: StrType}
+	case v == marray16:
+		return bytespec{size: 3, extra: array16v, typ: ArrayType}
+	case v == marray32:
+		return bytespec{size: 5, extra: array32v, typ: ArrayType}
+	case v == mmap16:
+		return bytespec{size: 3, extra: map16v, typ: MapType}
+	case v == mmap32:
+		return bytespec{size: 5, extra: map32v, typ: MapType}
+
+	// fixint
+	case v >= mfixint && v < 0x80:
+		return bytespec{size: 1, extra: constsize, typ: IntType}
+
+	// nfixint
+	case v >= mnfixint && uint16(v) < 0x100:
+		return bytespec{size: 1, extra: constsize, typ: IntType}
+
+	// fixstr gets constsize, since the prefix yields the size
+	case v >= mfixstr && v < 0xc0:
+		return bytespec{size: 1 + rfixstr(v), extra: constsize, typ: StrType}
+
+	// fixmap
+	case v >= mfixmap && v < 0x90:
+		return bytespec{size: 1, extra: varmode(2 * rfixmap(v)), typ: MapType}
+
+	// fixarray
+	case v >= mfixarray && v < 0xa0:
+		return bytespec{size: 1, extra: varmode(rfixarray(v)), typ: ArrayType}
+
+	}
+
+	panic(fatal)
```

Comment:
I can't tell if this case is reachable from the entry of this function, but we should take care that it isn't. Otherwise malformed input will lead to a panic here.

Might make sense to order these comparisons so that they are sequential (and therefore the last case of `uint16(v) < 0x100` is trivially true.

Reply:
agreed - fixed in the refactor